javascript/ql/src/semmle/javascript/frameworks/GraphQL.qll

```diff
@@ -0,0 +1,100 @@
+/**
+ * Provides classes for working with GraphQL connectors.
+ */
+
+import javascript
+
+/** Provides classes modelling concepts of GraphQL connectors. */
+module GraphQL {
+  /** A string-valued expression that is interpreted as a GraphQL query. */
+  abstract class GraphQLString extends DataFlow::Node { }
+}
+
+/**
+ * Provides classes modelling the octokit packages [@octokit/core](https://npmjs.com/package/@octokit/core),
+ * [@octokit/graphql](https://npmjs.com/package/@octokit/graphql), [@octokit/rest](https://npmjs.com/package/@octokit/rest),
+ * and [@octokit/request](https://npmjs.com/package/@octokit/request).
+ */
+private module Octokit {
+  /** Get an instanceof of `Octokit` */
+  private API::Node octokit() {
+    result =
+      API::moduleImport(["@octokit/core", "@octokit/rest"]).getMember("Octokit").getInstance()
```

Comment:
Related: https://github.com/actions/github-script magically expose such an instance as a (seemingly global) `github` variable. The source code is however in a `script` property of a yaml file, so we are unlikely to handle it at the moment.

Reply:
I think I'll just ignore that for now, given that we don't extract the relevant source code.